packages/commerce-sdk-react/src/hooks/useAccessToken.ts

```diff
@@ -23,4 +23,10 @@ const useAccessToken = (): string | null => {
     return auth.get('access_token')
 }
 
+export const useAccessTokenAsync = async () => {
+    const auth = useAuthContext()
+    const {access_token} = await auth.ready()
+    return access_token
+}
```

Comment:
I'm pretty sure you can't have async React hook, if you can, it would block the rendering of the components, right?

Reply:
I'm just thinking about `useAccessToken` v.s. `useAccessTokenAsync`.. I feel like as an end user, I always want to use `useAccessTokenAsync`, there is no reason to have `useAccessToken`

Reply:
what if we change the `useAccessToken` to return the `ready` promise by default, and remove `useAccessTokenAsync`

Reply:
dumb ideas...

```suggestion
export const useAccessTokenAsync = async () => {
    const auth = useAuthContext()
    return [auth.get('access_token'), auth.ready]
}
```

This way, you have access to both the token declaratively, and also the promise in case if you need to use it in action handlers.